### PR TITLE
fix for #272 improvements regarding Object Search

### PR DESCRIPTION
**Pull request by ashitsalesforce**
Changes to address the following:

- the ability to search both by Label or API Name (currently it works only for API Name)

- narrow down search results immediately, without the need to press any keys

### CODE REVIEW

**Review by reubencornel — APPROVED** [review]

File: src/main/java/com/salesforce/dataloader/ui/entitySelection/EntityFilter.java [hunk 1]
```diff
@@ -70,19 +72,20 @@ public boolean select(Viewer arg0, Object arg1, Object arg2) {
...
     }
     
-    private boolean filterBySearchText(String entityName) {
+    private boolean filterBySearchText(String entityName, String entityLabel) {
         String searchText = this.searchText.getText();
         if (searchText != null && !searchText.isEmpty() && entityName != null) {
             searchText = searchText.toLowerCase();
-            if (entityName.toLowerCase().contains(searchText)) {
+            if (entityName.toLowerCase().contains(searchText) 
+                    || entityLabel.toLowerCase().contains(searchText)) {
```
Comment (lzemskov) [inline src/main/java/com/salesforce/dataloader/ui/entitySelection/EntityFilter.java:88]: Do we also need a null check for entityLabel?

**Review by shireennagdive — APPROVED** [review]

**Comment by ashitsalesforce** [comment]
good point. I will make the change.

File: src/main/java/com/salesforce/dataloader/ui/entitySelection/EntityFilter.java (hunk 1 above)
Comment (ashitsalesforce) [reply src/main/java/com/salesforce/dataloader/ui/entitySelection/EntityFilter.java]: Done. please review.

Comment (reubencornel) [reply src/main/java/com/salesforce/dataloader/ui/entitySelection/EntityFilter.java]: There is no need for a null check here, it does not hurt, but its not required. This value comes from the describe result so it cannot be null.

**Pull request merged**